### ISSUE TITLE
Podman seems to conflate network errors with configuration errors

### ISSUE THREAD

**Issue by crawford** [containers/libpod#1451]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> kind bug

**Description**

When running Podman early in the boot process, I see the following error:

```
unable to find image: unable to pull quay.io/coreos/kube-client-agent:678cc8e6841e2121ebfdb6e2db568fce290b67d6: unable to find image in the registries defined in "/etc/containers/registries.conf"
```

If I re-run podman a little later, it works. I haven't modified `registries.conf` in the meantime.

**Steps to reproduce the issue:**

```
[core@crawford-libvirt-master-0 ~]$ systemctl status etcd-member-tls
● etcd-member-tls.service - etcd (System Application Container) TLS assets
   Loaded: loaded (/etc/systemd/system/etcd-member-tls.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2018-09-12 18:02:29 UTC; 1min 41s ago
  Process: 847 ExecStartPre=/bin/podman run --rm --volume /etc/ssl/etcd:/etc/ssl/etcd:rw,z --network host ${SIGNER_IMAGE} request --orgname=system:etcd-servers --cacrt=/etc/ssl/etcd/root-ca.crt --assetsdir=/etc/ssl/etcd --address=https://crawford-libvirt-api.local:6443 --dnsnames=localhost,*.kube-etcd.kube-system.svc.cluster.local,kube-etcd-client.kube-system.svc.cluster.local,crawford-libvirt-etcd-0.local --commonname=system:etcd-server:crawford-libvirt-etcd-0.local --ipaddrs=127.0.0.1 (code=exited, status=125)

Sep 12 18:02:29 localhost.localdomain systemd[1]: Starting etcd (System Application Container) TLS assets...
Sep 12 18:02:29 localhost.localdomain podman[847]: Trying to pull quay.io/coreos/kube-client-agent:678cc8e6841e2121ebfdb6e2db568fce290b67d6...Failed
Sep 12 18:02:29 localhost.localdomain podman[847]: unable to find image: unable to pull quay.io/coreos/kube-client-agent:678cc8e6841e2121ebfdb6e2db568fce290b67d6: unable to find image in the registries defined in "/etc/containers/registries.conf"
Sep 12 18:02:29 localhost.localdomain systemd[1]: etcd-member-tls.service: control process exited, code=exited status=125
Sep 12 18:02:29 localhost.localdomain systemd[1]: Failed to start etcd (System Application Container) TLS assets.
Sep 12 18:02:29 localhost.localdomain systemd[1]: Unit etcd-member-tls.service entered failed state.
Sep 12 18:02:29 localhost.localdomain systemd[1]: etcd-member-tls.service failed.
```
```
[core@crawford-libvirt-master-0 ~]$ sudo systemctl restart etcd-member-tls
```
```
[core@crawford-libvirt-master-0 ~]$ systemctl status etcd-member-tls
● etcd-member-tls.service - etcd (System Application Container) TLS assets
   Loaded: loaded (/etc/systemd/system/etcd-member-tls.service; static; vendor preset: enabled)
   Active: inactive (dead) since Wed 2018-09-12 18:04:27 UTC; 1s ago
  Process: 1677 ExecStart=/bin/true (code=exited, status=0/SUCCESS)
  Process: 1674 ExecStartPre=/bin/chown etcd:etcd /etc/ssl/etcd/system:etcd-peer:crawford-libvirt-etcd-0.local.key (code=exited, status=0/SUCCESS)
  Process: 1671 ExecStartPre=/bin/chown etcd:etcd /etc/ssl/etcd/system:etcd-peer:crawford-libvirt-etcd-0.local.crt (code=exited, status=0/SUCCESS)
  Process: 1668 ExecStartPre=/bin/chown etcd:etcd /etc/ssl/etcd/system:etcd-server:crawford-libvirt-etcd-0.local.key (code=exited, status=0/SUCCESS)
  Process: 1665 ExecStartPre=/bin/chown etcd:etcd /etc/ssl/etcd/system:etcd-server:crawford-libvirt-etcd-0.local.crt (code=exited, status=0/SUCCESS)
  Process: 1585 ExecStartPre=/bin/podman run --rm --volume /etc/ssl/etcd:/etc/ssl/etcd:rw,z --network host ${SIGNER_IMAGE} request --orgname=system:etcd-peers --cacrt=/etc/ssl/etcd/root-ca.crt --assetsdir=/etc/ssl/etcd --address=https://crawford-libvirt-api.local:6443 --dnsnames=*.kube-etcd.kube-system.svc.cluster.local,kube-etcd-client.kube-system.svc.cluster.local,crawford-libvirt-etcd-0.local --commonname=system:etcd-peer:crawford-libvirt-etcd-0.local (code=exited, status=0/SUCCESS)
  Process: 1486 ExecStartPre=/bin/podman run --rm --volume /etc/ssl/etcd:/etc/ssl/etcd:rw,z --network host ${SIGNER_IMAGE} request --orgname=system:etcd-servers --cacrt=/etc/ssl/etcd/root-ca.crt --assetsdir=/etc/ssl/etcd --address=https://crawford-libvirt-api.local:6443 --dnsnames=localhost,*.kube-etcd.kube-system.svc.cluster.local,kube-etcd-client.kube-system.svc.cluster.local,crawford-libvirt-etcd-0.local --commonname=system:etcd-server:crawford-libvirt-etcd-0.local --ipaddrs=127.0.0.1 (code=exited, status=0/SUCCESS)
 Main PID: 1677 (code=exited, status=0/SUCCESS)

Sep 12 18:04:22 crawford-libvirt-master-0 podman[1486]: Copying blob sha256:81033e7c1d6a5b44a94bb6b40033a6e589f50fd6b61578da6fc809e61f83898d
Sep 12 18:04:23 crawford-libvirt-master-0 podman[1486]: [99B blob data]
Sep 12 18:04:23 crawford-libvirt-master-0 podman[1486]: Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Sep 12 18:04:23 crawford-libvirt-master-0 podman[1486]: [28B blob data]
Sep 12 18:04:23 crawford-libvirt-master-0 podman[1486]: Copying blob sha256:c9daf76118657aaec02ae99978e42c6a2ba13627f29dc3ed981ed73a45a5e9f1
Sep 12 18:04:24 crawford-libvirt-master-0 podman[1486]: [97B blob data]
Sep 12 18:04:24 crawford-libvirt-master-0 podman[1486]: Skipping fetch of repeat blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Sep 12 18:04:24 crawford-libvirt-master-0 podman[1486]: Writing manifest to image destination
Sep 12 18:04:24 crawford-libvirt-master-0 podman[1486]: Storing signatures
Sep 12 18:04:27 crawford-libvirt-master-0 systemd[1]: Started etcd (System Application Container) TLS assets.
```
```
[core@crawford-libvirt-master-0 ~]$ cat /etc/containers/registries.conf 
# This is a system-wide configuration file used to
# keep track of registries for various container backends.
# It adheres to TOML format and does not support recursive
# lists of registries.

# The default location for this configuration file is /etc/containers/registries.conf.

# The only valid categories are: 'registries.search', 'registries.insecure', 
# and 'registries.block'.

[registries.search]
registries = ['registry.access.redhat.com']

# If you need to access insecure registries, add the registry's fully-qualified name.
# An insecure registry is one that does not have a valid SSL certificate or only does HTTP.
[registries.insecure]
registries = []


# If you need to block pull access from a registry, uncomment the section below
# and add the registries fully-qualified name.
#
# Docker only
[registries.block]
registries = []
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 0.7.3
```

**Output of `podman info`:**

```
host:
  MemFree: 1623007232
  MemTotal: 2096586752
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 2
  hostname: crawford-libvirt-master-0
  kernel: 3.10.0-862.14.1.el7.x86_64
  os: linux
  uptime: 10m 15.3s
insecure registries:
  registries: []
registries:
  registries:
  - registry.access.redhat.com
store:
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 1
  RunRoot: /var/run/containers/storage
```

**Comment by baude** [comment]
This is fixed in master.  The error is more clear.

**Comment by wking** [comment]
> This is fixed in master.

Can you link the fix?  I'm hunting around, but haven't turned it up yet.

Is this a dup of #1212, or is that different?

**Comment by wking** [comment]
> > This is fixed in master.
>
> Can you link the fix?

Ah, grepping for the error message turned up #1194 and #1195.  I'm guessing you meant #1195, which landed in v0.8.1.  I've also just filed #1456 as another incremental improvement, although as @rhatdan [mentions][1] it would be nice if it was easier to address both this and projectatomic/buildah#849 at the same time.

[1]: https://github.com/containers/libpod/issues/1212#issuecomment-412958373

**Comment by wking** [comment]
> If I re-run podman a little later, it works. I haven't modified `registries.conf` in the meantime.

#1456 has addressed the conflation.  It won't have fixed whatever caused your first call to fail, but the new error messages should now be more useful.  Should we close this and open a new issue once you get a new error message with this failure?  Or leave this open and use it once you get a new error message?
